controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -107,237 +107,70 @@ func ReadEnv() string {
 	return cEnv
 }
 
-//Reconcile fn
-func (r *ClowdEnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+//Acts as a guard for a reconciliation cycle, as well as initatizes a bunch of required objects
+func (r *ClowdEnvironmentReconciler) initReconciliation(ctx context.Context, req ctrl.Request) (logr.Logger, context.Context, crd.ClowdEnvironment, ctrl.Result, error) {
 	log := r.Log.WithValues("env", req.Name).WithValues("rid", utils.RandString(5))
 	ctx = context.WithValue(ctx, errors.ClowdKey("log"), &log)
 	ctx = context.WithValue(ctx, errors.ClowdKey("recorder"), &r.Recorder)
 	env := crd.ClowdEnvironment{}
 
+	silentFailError := errors.New(SILENTFAIL)
+
 	if getEnvErr := r.Client.Get(ctx, req.NamespacedName, &env); getEnvErr != nil {
 		if k8serr.IsNotFound(getEnvErr) {
 			// Must have been deleted
-			return ctrl.Result{}, nil
+			return log, ctx, env, ctrl.Result{}, silentFailError
 		}
 		log.Info("Namespace not found", "err", getEnvErr)
-		return ctrl.Result{}, getEnvErr
+		return log, ctx, env, ctrl.Result{}, getEnvErr
 	}
 
+	return log, ctx, env, ctrl.Result{}, nil
+}
+
+func (r *ClowdEnvironmentReconciler) initMetrics(env crd.ClowdEnvironment) {
 	if _, ok := presentEnvironments[env.Name]; !ok {
 		presentEnvironments[env.Name] = true
 	}
 	presentEnvsMetric.Set(float64(len(presentEnvironments)))
 
 	delete(managedEnvironments, env.Name)
+}
 
-	defer func() {
-		managedEnvsMetric.Set(float64(len(managedEnvironments)))
-	}()
+//Reconcile fn
+func (r *ClowdEnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log, ctx, env, res, err := r.initReconciliation(ctx, req)
```

Comment:
A few things around this
1) I'm not sure I understand why we are doing it differently to the clowdapp
2) Passing in the context and getting a modified version back seems rather strange to me
I propose making it like ClowdApp

Reply:
Done